cmake/FindLibserialport.cmake

```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: MIT
+
+#[=======================================================================[.rst:
+FindLibserialport
+-------
+
+Finds the sigrok serial port library (``libserialport``)
```

Comment:
1. Fair, wasn't sure if there were any other libserialport libraries since it felt like a pretty common name.
2. I don't really get what you're trying to say. In CMake, using a find script is the normal way to link to other libraries. With a find script, standard errors are given when the library isn't found, the call site is much cleaner, and this script can be reused in other projects.